src/recipes/serializers.py

```diff
@@ -0,0 +1,43 @@
+from rest_framework import serializers
+from rest_framework.exceptions import ValidationError
+
+from .models import Recipe, Ingredient
+
+
+class IngredientSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Ingredient
+        fields = ['name']
+
+
+class RecipeSerializer(serializers.ModelSerializer):
+    """
+    A `Recipe` is a list of ingredients with a name and description.
+    """
+    ingredients = IngredientSerializer(many=True)
+
+    class Meta:
+        model = Recipe
+        fields = ['id', 'name', 'description', 'ingredients']
+
+    def create(self, validated_data) -> Recipe:
+        ingredients_data = validated_data.pop('ingredients')
+        recipe = Recipe.objects.create(**validated_data)
+        for ingredient_data in ingredients_data:
+            Ingredient.objects.create(recipe=recipe, **ingredient_data)
+        return recipe
+
+    def update(self, instance, validated_data) -> Recipe:
+        ingredients_data = validated_data.pop('ingredients', [])
+        recipe = super().update(instance, validated_data)
+        Ingredient.objects.filter(recipe=recipe.id).delete()
```

Comment:
Could be wrong, but I'd say `filter(recipe=recipe)` works as well, Django should handle this under the hood.

Reply:
Good point. I was getting an error in this function. I fixed a bunch of things and may have neglected to change this back.